flink-core/src/main/java/org/apache/flink/api/common/eventtime/WatermarkGeneratorSupplier.java

```diff
@@ -54,5 +55,14 @@ interface Context {
          * @see MetricGroup
          */
         MetricGroup getMetricGroup();
+
+        /**
+         * Returns a {@link RelativeClock} that hides periods when input was not active and {@link
+         * WatermarkGenerator} could not have been executed due to execution being blocked by the
+         * runtime. For example a backpressure or watermark alignment blocking the progress.
+         *
+         * @see RelativeClock
+         */
+        RelativeClock getInputActivityClock();
```

Comment:
Hmm..why doesn't this break compatibility in 1.20?

Reply:
I don't know. Maybe someone forgot a release step?